sdk/eventhub/Azure.Messaging.EventHubs.Processor/src/EventProcessorClient.cs

```diff
@@ -949,7 +950,7 @@ private async Task RunAsync(CancellationToken cancellationToken)
                     // When resuming from a checkpoint, the intent to process the next available event in the stream which
                     // follows the one that was used to create the checkpoint.  Create the position using an exclusive offset.
 
-                    startingPosition = EventPosition.FromOffset(checkpoint.Offset, false);
+                    startingPosition = EventPosition.FromOffset(checkpoint.Offset.GetValueOrDefault(0), false);
```

Comment:
Yup, I left it out because we're planning on refactoring this to be based on the Event Processor, but I'll throw in a basic version here, just in case.